src/main/java/org/motechproject/prevac/helper/DtoLookupHelper.java

```diff
@@ -107,30 +108,41 @@ public static GridSettings changeLookupForVisitReschedule(GridSettings settings)
             fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN);
         } else {
             fieldsMap = getFields(settings.getFields());
-            switch (settings.getLookup()) {
-                case "Find By Visit Type":
-                    String type = (String) fieldsMap.get(Visit.VISIT_TYPE_PROPERTY_NAME);
-                    if (StringUtils.isBlank(type) || !AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN.contains(VisitType.valueOf(type))) {
-                        fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, null);
-                    }
-                    settings.setLookup(settings.getLookup() + " And Planned Date");
-                    break;
-                default:
-                    fieldsMap.put(Visit.VISIT_TYPE_PROPERTY_NAME, AVAILABLE_VISIT_TYPES_FOR_RESCHEDULE_SCREEN);
-                    settings.setLookup(settings.getLookup() + " And Visit Type Set And Planned Date");
-                    break;
-            }
         }
 
         Map<String, String> rangeMap = getDateRangeFromFilter(settings);
-
-        if (rangeMap != null && (StringUtils.isNotBlank(rangeMap.get("min")) || StringUtils.isNotBlank(rangeMap.get("max")))) {
-            settings.setLookup(settings.getLookup() + " Range");
-            fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, rangeMap);
-        } else {
-            fieldsMap.put(Visit.VISIT_PLANNED_DATE_PROPERTY_NAME, null);
+        String lookup = settings.getLookup();
+
+        switch (lookup) {
+            case "Find By Visit Type And Planned Date Range":
+                break;
+            case "Find By Visit Type Set And Planned Date":
```

Comment:
this lookup is not in the available lookups, so it will never be used

Reply:
This lookup is set when there is no lookup selected on UI. Then in this case I check if there are any data in date filter and if so I modify lookup to handle date range.

Reply:
ok, it's fine then